### PR TITLE
kv: merge batch get results to fix insert ignore in a transaction block

### PR DESCRIPTION
**Pull request by jackysp**
Merge the batch get results with buffer store to fix insert ignore bug when it is in a transaction block.
PTAL @coocood

### CODE REVIEW

**Comment by coocood** [comment]
@jackysp 
If transaction `IsReadOnly`, we can by pass the merging.

File: kv/txn.go [hunk 1]
```diff
@@ -88,3 +88,31 @@ func BackOff(attempts uint) int {
 	time.Sleep(sleep)
 	return int(sleep)
 }
+
+// BatchGetValues gets values in batch.
+// The values from buffer in transaction and the values from the storage node are merged together.
+func BatchGetValues(txn Transaction, keys []Key) (map[string][]byte, error) {
+	bufferValues := make(map[string][]byte, len(keys))
```
Comment (coocood) [inline kv/txn.go]: Using a keys slice and a values slice has a lower cost.

**Comment by jackysp** [comment]
PTAL @coocood

**Comment by coocood** [comment]
LGTM

File: kv/txn.go [hunk 2]
```diff
@@ -88,3 +88,37 @@ func BackOff(attempts uint) int {
...
+	storageValues, err := txn.GetSnapshot().BatchGet(shrinkKeys)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	for i, key := range keys {
+		if bufferValues[i] == nil {
+			continue
+		}
+		storageValues[string(key)] = bufferValues[i]
+	}
```
Comment (zimulala) [inline kv/txn.go]: Do we need to do `	CheckVisibility(txn.startTS)` ?

**Pull request merged**